### PR TITLE
Stop dispatching websocket messages once connection is known closed

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

File: ws/src/session.rs [hunk 1]
```diff
@@ -233,6 +233,10 @@ impl<M: core::Metadata, S: core::Middleware<M>> ws::Handler for Session<M, S> {
 	}
 
 	fn on_message(&mut self, msg: ws::Message) -> ws::Result<()> {
+		if !self.active.load(atomic::Ordering::SeqCst) {
+			return Err(ws::Error::new(ws::ErrorKind::Internal, "Attempting to send a message to closed/closing connection."))
+		}
```
Comment (rphmeier) [inline ws/src/session.rs]: not actually sending anything here, only receiving

Comment (NikVolf) [inline ws/src/session.rs]: yeah, moved to a handling future

File: ws/src/server_builder.rs [hunk 2]
```diff
@@ -54,10 +57,11 @@ impl ::std::error::Error for Error {
...
 
     fn cause(&self) -> Option<&::std::error::Error> {
-        Some(match *self {
-            Error::Io(ref e) => e,
-            Error::WebSocket(ref e) => e,
-        })
+        match *self {
+            Error::Io(ref e) => Some(e),
+            Error::WebSocket(ref e) => Some(e),
+			Error::ConnectionClosed => None,
```
Comment (ngotchac) [inline ws/src/server_builder.rs]: There might be a space/tab mismatch here

Comment (rphmeier) [inline ws/src/server_builder.rs]: yea, it uses tabs like the rest of the codebase while the file (mistakenly) uses spaces

Comment (NikVolf) [inline ws/src/server_builder.rs]: see https://github.com/paritytech/jsonrpc/pull/177

**Pull request merged**